wordpress-mobile/WordPress-Android
# [Bloganuary] Modal analytics tracking

Part of #19685

Add tracking for the Bloganuary Learn More model based on specs (pctCYC-16J-p2).

-----

## To Test:

1. Install and log into the app with a site that has Blogging Prompts available
2. Go to the My Site dashboard
3. Tap "Learn More" on the Bloganuary card
4. **Verify**, in the logs, `🔵 Tracked: bloganuary_nudge_learn_more_modal_shown, Properties: {"prompts_enabled":[true if prompts are enabled, false otherwise]}`
8. Tap the CTA action
9. **Verify**, in the logs, `🔵 Tracked: bloganuary_nudge_learn_more_modal_action_tapped, Properties: {"action":["dismiss" or "turn_on_prompts", depending if prompts are enabled or not]}`

Modal dismissal:
Whenever the Modal is dismissed (no matter which action led to it), there should be tracking for dismissal. Test multiple dismiss methods (back button, close button, fling down, tap CTA) and **verify** `🔵 Tracked: bloganuary_nudge_learn_more_modal_dismissed` in the app logs.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - Analytics tracker unit tests.

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
No UI changes.
